### PR TITLE
added functionality for optionally adding principalId to an auto-tag policy

### PR DESCRIPTION
**Pull request by JohnTheodore**
I added a doc use case for how to do auto-tagging too.

### CODE REVIEW

**Comment by kapilt** [comment]
would you want this in preference to the username or in addition to it. just wondering if we need a user_expr to extract from the request instead of trying to support both as separate values.

**Comment by kapilt** [comment]
re behavior, default in the context of extant would be current default (expression unspecified), as it has special handling, but else we utilize a non default value as literal jmespath expression result.

**Comment by JohnTheodore** [comment]
note, I think the 'root' user gets skipped in this autotagger logic by default. and ASGs can use the root user by default, which complicates some things. In the coming weeks we'll test out the auto-tagging more and there might be more PRs to try and solve some corner cases.

but I'll keep this PR just opt-in principalId to keep it small.

I would want this in addition to the username. We use cloudhealth to let users view their spend (whic his why we want to auto-tag everything). So alice can just lookup all the AWS costs for CreatorName alice. but principalId (CreatorId), we use that to enforce IAM. So if Bob spins up an EC2 instance, Alice won't be able to shut down that instance since Alice's CreatorId (principalId) doesn't match the EC2 instance tag value. so we try not to spin up endless AWS accounts, and we want to share aws accounts across different teams. but we don't want those teams to step on each others toes too much. docs for what I'm describing are here: https://aws.amazon.com/blogs/security/how-to-automatically-tag-amazon-ec2-resources-in-response-to-api-events/

File: c7n/actions.py [hunk 1]
```diff
@@ -557,24 +560,36 @@ def process(self, resources, event):
...
 
         tag_action = self.manager.action_registry.get('tag')
-        tag_action(
-            {'key': self.data['tag'], 'value': user},
-            self.manager).process(untagged)
-        return untagged
+        new_tags = {
+            self.data['tag']: user
+        }
+        principal_id_key = self.data.get('principal_id_tag', None)
```
Comment (kapilt) [inline c7n/actions.py]: this line and the line just below are duplicates.
